code/datums/martial/psychotic_brawl.dm

```diff
@@ -3,42 +3,156 @@
+			addtimer(CALLBACK(src, .proc/pounce, A, D), 20)
+			D.visible_message("<span class='warning'>[A] crouches down, preparing to pounce at [D]!</span>", "<span class='userdanger'>[A] prepares to pounce at you!</span>")
+		if(4)
 			if(A.grab_state >= GRAB_AGGRESSIVE)
 				D.grabbedby(A, 1)
 			else
 				A.start_pulling(D, supress_message = TRUE)
 				if(A.pulling)
 					D.drop_all_held_items()
 					D.stop_pulling()
-					if(A.a_intent == INTENT_GRAB)
-						log_combat(A, D, "grabbed", addition="aggressively")
-						D.visible_message("<span class='warning'>[A] violently grabs [D]!</span>", \
-										"<span class='userdanger'>You're violently grabbed by [A]!</span>", "<span class='hear'>You hear sounds of aggressive fondling!</span>", null, A)
-						to_chat(A, "<span class='danger'>You violently grab [D]!</span>")
-						A.setGrabState(GRAB_AGGRESSIVE) //Instant aggressive grab
-					else
-						log_combat(A, D, "grabbed", addition="passively")
-						A.setGrabState(GRAB_PASSIVE)
-		if(4)
+					log_combat(A, D, "grabbed", addition="aggressively")
+					D.visible_message("<span class='warning'>[A] violently grabs [D]!</span>", \
+									"<span class='userdanger'>You're violently grabbed by [A]!</span>", "<span class='hear'>You hear sounds of aggressive fondling!</span>", null, A)
+					to_chat(A, "<span class='danger'>You violently grab [D]!</span>")
+					A.setGrabState(GRAB_AGGRESSIVE) //Instant aggressive grab
+		if(5,6)
+			atk_verb = "grabbed"
+			A.start_pulling(D)
+	if(atk_verb)
+		log_combat(A, D, "[atk_verb] (Psychotic Brawling)")
+	return TRUE
+
+/datum/martial_art/psychotic_brawling/proc/pounce(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	if(A.stat)
+		return
+	if(A.throw_at(D.loc, 7, 2, force = MOVE_FORCE_VERY_STRONG))
+		if(!(D.mobility_flags & MOBILITY_STAND))
+			A.throw_at(D.loc, 2, 4, spin = FALSE, force = MOVE_FORCE_VERY_WEAK)
+			A.SetKnockdown(0)
+			A.SetImmobilized(0)
+			A.SetParalyzed(0)
+			A.start_pulling(D, supress_message = TRUE)
+			if(A.pulling)
+				D.drop_all_held_items()
+				D.stop_pulling()
+				log_combat(A, D, "grabbed", addition="aggressively")
+				D.visible_message("<span class='warning'>[A] pounces onto [D]!</span>", "<span class='userdanger'>[A] grabs hold of your neck!</span>")
+				to_chat(A, "<span class='danger'>You grab [D] by the neck!</span>")
+				A.setGrabState(GRAB_NECK)
+
+/datum/martial_art/psychotic_brawling/proc/psycho_attack(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	var/atk_verb
+	var/armor_block = D.run_armor_check(attack_flag = "melee")
+	switch(rand(1,6))
+		if(1)
+			playsound(A, 'sound/weapons/bite.ogg', 50, 1, -1)
+			A.do_attack_animation(D, ATTACK_EFFECT_BITE)
+			D.apply_damage(D.dna.species.punchdamage-3, A.dna.species.attack_type, blocked = armor_block)
+			for(var/datum/disease/V in A.diseases)
+				if((V.spread_flags & DISEASE_SPREAD_SPECIAL) || (V.spread_flags & DISEASE_SPREAD_NON_CONTAGIOUS) || (V.spread_flags & DISEASE_SPREAD_FALTERED))
```

Comment:
```suggestion
				if((V.spread_flags &(DISEASE_SPREAD_SPECIAL|DISEASE_SPREAD_NON_CONTAGIOUS|DISEASE_SPREAD_FALTERED))
```

Reply:
Doesnt this check if it has all of these flags, instead of just one?

Reply:
Nah, it checks for any

Reply:
It only needs one